### ISSUE TITLE
Testing ansible idempotency

### ISSUE THREAD

**Issue by bborysenko** [neillturner/kitchen-ansible#116]
Is there are some way to do it with kitchen-ansible ?

**Comment by metmajer** [comment]
Generally, there's no test for idempotency, unless you define a test scenario that allows you to test idempotent behavior of your module(s). As you probably know, a test case in kitchen-ansible is an ansible playbook that is located somewhere under test/integration. In such a test case you would call your role multiple times and then use serverspec (or any other of the supported test frameworks) to test whether the state has only been enforced once. If you think about an entry in a configuration file, you could test, whether the entry was created only once. Does that make sense?

**Comment by rndmh3ro** [comment]
To follow up on @metmajer, here's the way I do it:
```
- name: wrapper playbook for kitchen testing "ansible-os-hardening" with custom vars for testing
  hosts: localhost
  roles:
    - ansible-os-hardening

- name: wrapper playbook for kitchen testing "ansible-os-hardening" with custom vars for testing
  hosts: localhost
  roles:
    - ansible-os-hardening
```

This essentially runs the role on the same host twice, so I can see if anything changed in the second run.

Though it would be nice to have an idempotency test like https://github.com/ahelal/kitchen-ansiblepush.

**Comment by bborysenko** [comment]
@metmajer I suppose that serverspec or some other test framework is not well suitable for that kind, since it's require to make sure that we have idempotency test suite for each ansible task. 

@rndmh3ro thanks for the link! Having something similar would be really great.

**Comment by neillturner** [comment]
thanks for PR in version 0.0.34
